doc/DOCUMENTATION.org

```diff
@@ -542,6 +543,58 @@ configuration at the beginning and end of =Spacemacs= loading process.
     loading.
   - =dotspacemacs/config= is triggered at the very end of =Spacemacs= loading.
 
+*** Binding keys
+Key sequences are bound to commands in Emacs in various keymaps. The most basic
+map is the global-map. Setting a key binding the global-map uses the function
+=global-set-key= as follows (to the command =forward-char= in this case).
+
+#+begin_src emacs-lisp
+  (global-set-key (kbd "C-]") 'forward-char)
+#+end_src
+
+The =kbd= macro accepts a string describing a key sequence. The global-map is
+often shadowed by other maps. For example, evil-mode defines keymaps that target
+states (or modes in vim terminology). Here is an example that creates the same
+binding as above but only in insert state (=define-key= is a built-in function.
+Evil-mode has its own functions for defining keys).
+
+#+begin_src emacs-lisp
+  (define-key evil-insert-state-map (kbd "C-]") 'forward-char)
+#+end_src
+
+Perhaps most importantly for spacemacs is the use of the evil-leader package,
+which binds keys to the evil-leader keymap. This is where most of the spacemacs
+bindings live. There are two related commands from this package which are used
+as follows.
+
+#+begin_src emacs-lisp
+  (evil-leader/set-key "C-]" 'forward-char)
+  (evil-leader/set-key-for-mode 'emacs-lisp-mode "C-]" 'forward-char)
+#+end_src
+
+These functions use a macro like =kbd= to translate the key sequences for you.
+The second function, =evil-leader/set-key-for-mode=, binds the key only in the
+specified mode. The second key binding would not be in effect in =org-mode= for
+example.
+
+Finally, one should be aware of prefix keys. Essentially, all keymaps can be
+nested. Nested keymaps are used extensively in spacemacs, and in vanilla Emacs
+for that matter. For example, ~SPC a~ points to key bindings for "applications",
+like ~SPC ac~ for =calc-dispatch=. Nesting bindings is easy.
+
+#+begin_src emacs-lisp
+  (spacemacs/declare-prefix "]" "bracket-prefix")
+  (evil-leader/set-key "]]" 'double-bracket-command)
+#+end_src
```

Comment:
@mbertheau Why don't you open up a pr for this, so we can discuss it there? I'm happy to provide more explanation in the docs about what's going on. I just wasn't sure where to draw the line so to speak. 

You're right that declaring the prefix only serves to name the prefix and is unnecessary for the functionality. That's why I was saying it's easy. 

`spacemacs/declare-prefix` was recently changed to support the new way of naming prefixes, which is entirely done through which-key, but it also supports the old way, which I hope to phase out eventually. The old way actually renamed the keymap's symbol (which is why the name had to be unique, which was causing problems, etc) and was more of a hack to be honest. The new way just tells keymap that this key binding is a prefix and declares a name for it. Again, I'm happy to provide more detail, and please feel free to rework the explanation in a new pr since this has been closed already.